src/test/storage/dictionary_segment_test.cpp

```diff
@@ -3,68 +3,145 @@
+  vc_int->append(10);
+  vc_int->append(10);
+
+  auto dict_col = std::make_shared<DictionarySegment<int>>(vc_int);
+
+  // values:                  0, 2, 4, 6, 8, 10, 10, 10
+  // dictionary size:         6 * sizeof(int)
+  // attribute vector size:   8 * sizeof(uint8_t)
+  EXPECT_EQ(dict_col->estimate_memory_usage(), 6 * sizeof(int) + 8 * sizeof(uint8_t));
+}
+
+TEST_F(StorageDictionarySegmentTest, BracketOperator) {
+  vc_str->append("World");
+  vc_str->append("Hallo");
+  vc_str->append("World");
+
+  auto dict_col = std::make_shared<DictionarySegment<std::string>>(vc_str);
+
+  EXPECT_EQ(dict_col->operator[](1), AllTypeVariant{"Hallo"});
+}
+
+TEST_F(StorageDictionarySegmentTest, AppendNotPossible) {
+  auto dict_col = std::make_shared<DictionarySegment<std::string>>(vc_str);
+
+  EXPECT_THROW(dict_col->append(AllTypeVariant{"Hallo"}), std::runtime_error);
+}
+
+TEST_F(StorageDictionarySegmentTest, CorrectNumberOfBits) {
+  auto dictionary_segment_fixture = std::make_shared<DictionarySegmentFixture<int>>(vc_int);
+
+  int minimal_number8 = dictionary_segment_fixture->get_minimal_number_of_bits_for_dictionary_size(
+      std::numeric_limits<uint8_t>::max() - 1);
+  int minimal_number16 = dictionary_segment_fixture->get_minimal_number_of_bits_for_dictionary_size(
+      std::numeric_limits<uint16_t>::max() - 1);
+  int minimal_number32 = dictionary_segment_fixture->get_minimal_number_of_bits_for_dictionary_size(
+      std::numeric_limits<uint32_t>::max() - 1);
+  int minimal_number64 = dictionary_segment_fixture->get_minimal_number_of_bits_for_dictionary_size(
+      std::numeric_limits<uint64_t>::max() - 1);
+
+  EXPECT_EQ(minimal_number8, 8);
+  EXPECT_EQ(minimal_number16, 16);
+  EXPECT_EQ(minimal_number32, 32);
+  EXPECT_EQ(minimal_number64, 64);
+}
+
+TEST_F(StorageDictionarySegmentTest, CreateAttributeVector) {
+  auto dictionary_segment_fixture = std::make_shared<DictionarySegmentFixture<int>>(vc_int);
+  std::vector<int> dictionary_vector{1, 2, 3};
+  auto dictionary = std::make_shared<std::vector<int>>(dictionary_vector);
+
+  vc_int->append(2);
+  vc_int->append(3);
+  vc_int->append(1);
+  vc_int->append(1);
+
+  int minimal_bytes = 16;
+  auto attribute_vector =
+      dictionary_segment_fixture->create_attribute_vector_with_type(minimal_bytes, vc_int, dictionary);
+
+  EXPECT_EQ(attribute_vector->width(), 2);
+  EXPECT_EQ(attribute_vector->size(), 4);
+  EXPECT_EQ(attribute_vector->get(0), 1);
+  EXPECT_EQ(attribute_vector->get(1), 2);
+  EXPECT_EQ(attribute_vector->get(2), 0);
+  EXPECT_EQ(attribute_vector->get(3), 0);
+
+  minimal_bytes = 32;
+  attribute_vector = dictionary_segment_fixture->create_attribute_vector_with_type(minimal_bytes, vc_int, dictionary);
+  EXPECT_EQ(attribute_vector->width(), 4);
+
+  minimal_bytes = 64;
+  attribute_vector = dictionary_segment_fixture->create_attribute_vector_with_type(minimal_bytes, vc_int, dictionary);
+  EXPECT_EQ(attribute_vector->width(), 8);
+}
```

Comment:
Great tests. An additional test which tests the more integrated functionality would probably be helpful, for example: creating a dictionary segment with > 300 elements and then checking whether the resulting `attribute_vector` has the correct width and the resulting dictionary is correct as well.